### PR TITLE
C++ named namespaces support

### PR DESCRIPTION
**Pull request by Klaim**

### CODE REVIEW

File: asdl/lang/cpp/cpp_asdl.simplified.txt [hunk 1]
```diff
@@ -16,7 +16,7 @@ declaration = AccessSpecDecl(identifier access_spec)
             | FriendDecl(identifier type)
             | FunctionDecl(identifier name, type return_type, constant? storage, constant? variadic, constant? inline, exception_spec? exception, statement? body, parameter* parameters)
             | FunctionTemplateDecl(statement* subnodes)
-            | NamespaceDecl(identifier name, statement* subnodes)
+            | NamespaceDecl(identifier name, declaration* subnodes)
```
Comment (serge-sans-paille) [inline asdl/lang/cpp/cpp_asdl.simplified.txt:19]: `identifier? name` to support anonymous namespaces?

File: cpplang/parser.py [hunk 2]
```diff
@@ -933,7 +933,7 @@ def parse_CXXNullPtrLiteralExpr(self, node) -> tree.CXXNullPtrLiteralExpr:
     @parse_debug
     def parse_NamespaceDecl(self, node) -> tree.NamespaceDecl:
         assert node['kind'] == "NamespaceDecl"
-        name = node['name']
+        name = node.get('name', "") # Namespaces can be anonymous (no name given)
```
Comment (serge-sans-paille) [inline cpplang/parser.py:936]: just `node.get("name")` if you declare `identifier? name`

Comment (serge-sans-paille) [reply cpplang/parser.py:936]: You may need to adjust `code_gen.py` though.

Comment (Klaim) [reply cpplang/parser.py:936]: Yeah I ended up using empty-string and reverted my changes to `code_gen.py` as in the end it didnt need to change.
Not sure which way is best, both should work.

**Comment by Klaim** [comment]
Note that right now anonymous namespaces dont work but I didnt yet understand what was the issue exactly so I deactivated it for now while I'm investigating.

File: cpplang/parser.py (hunk 2 above)
Comment (serge-sans-paille) [reply cpplang/parser.py:936]: I think the `identifier? name` more cleanly converys the fact that the name is either an identifier, or no name.

Comment (Klaim) [reply cpplang/parser.py:936]: ok then, I'll go with that 👍🏽

**Comment by Klaim** [comment]
Not done yet, this revealed issues with how declarations are handle when the code is generated (basically `;` as missing when they should be there)

**Comment by serge-sans-paille** [comment]
@kleag: this LGTM, can you approve CI? Thanks!

**Comment by kleag** [comment]
Checks failed

**Comment by Klaim** [comment]
It should pass now 👍🏽

**Comment by kleag** [comment]
Why checks do not start automatically? Do I have to setup something?

**Comment by Klaim** [comment]
@kleag 
I believe there is a setting in the repo to require that all PRs against a branch (`main` usually) pass the ci - it's probably in the branch's settings.

**Comment by kleag** [comment]
> @kleag I believe there is a setting in the repo to require that all PRs against a branch (`main` usually) pass the ci - it's probably in the branch's settings.

Yes, but checks for PRs by @serge-sans-paille are executed with no action from my side and I don't see the difference

**Comment by kleag** [comment]
Found it. There is a setting requiring approval for first time contributors. I did not remember. Your future PRs should be able to start checks by themselves.

**Comment by Klaim** [comment]
Ah cool thanks!

**Pull request merged**